### PR TITLE
Added types and tweaked PHPdoc of clickLink() and submitForm() methods

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | -   
| License       | MIT
| Doc PR        | -

I wanted to suggest these changes to #27807 but I failed to do that before it was merged.

### CODE REVIEW

File: src/Symfony/Component/BrowserKit/Client.php [hunk 1]
```diff
@@ -322,18 +318,17 @@ public function submit(Form $form, array $values = array()/*, array $serverParam
...
-     * @param array  $values An array of form field values
-     * @param string $method The method for the form
-     *
-     * @return Crawler
+     * @param string $button      The text content, id, value or name of the form <button> or <input type="submit">
+     * @param array  $fieldValues Use this syntax: array('my_form[name]' => '...', 'my_form[email]' => '...')
+     * @param string $method      The HTTP method used to submit the form
      */
-    public function submitForm($button, $values, $method)
+    public function submitForm(string $button, array $fieldValues, string $method = 'POST'): Crawler
```
Comment (fabpot) [inline src/Symfony/Component/BrowserKit/Client.php]: What about having `array()` as the default value for `$fieldValues`?

Comment (javiereguiluz) [reply src/Symfony/Component/BrowserKit/Client.php]: I like that! Done!

**Comment by javiereguiluz** [comment]
Should we add to `submitform()` a fourth optional argument called `array $serverParameters = array()` like in `submit()`?

**Comment by fabpot** [comment]
Might be nice.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @javiereguiluz.

**Pull request merged**